dotCMS/src/integration-test/java/com/dotcms/filters/VanityUrlFilterTest.java

```diff
@@ -0,0 +1,180 @@
+package com.dotcms.filters;
+
+import java.io.File;
+import java.util.Collection;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import com.dotcms.datagen.SiteDataGen;
+import com.dotcms.mock.request.MockHttpRequest;
+import com.dotcms.mock.response.MockHeaderResponse;
+import com.dotcms.mock.response.MockHttpCaptureResponse;
+import com.dotcms.mock.response.MockHttpResponse;
+import com.dotcms.mock.response.MockHttpStatusResponse;
+import com.dotcms.util.FiltersUtil;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotcms.vanityurl.business.VanityUrlAPI;
+import com.dotcms.vanityurl.cache.VanityUrlCache;
+import com.dotcms.vanityurl.filters.VanityURLFilter;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.liferay.portal.model.User;
+import com.liferay.util.FileUtil;
+
+public class VanityUrlFilterTest {
+
+
+    private static ContentletAPI contentletAPI;
+    private static VanityUrlAPI vanityUrlAPI;
+    private static Host defaultHost;
+    private static User user;
+    private static Language defaultLanguage;
+    private static VanityUrlCache cache;
+    private static FiltersUtil filtersUtil;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+
+        /* Default user */
+        user = APILocator.systemUser();
+
+        contentletAPI = APILocator.getContentletAPI();
+        vanityUrlAPI = APILocator.getVanityUrlAPI();
+        filtersUtil = FiltersUtil.getInstance();
+
+        cache=CacheLocator.getVanityURLCache();
+
+        /* Default variables */
+        defaultHost =  new SiteDataGen().nextPersisted();
+        defaultLanguage = APILocator.getLanguageAPI().getDefaultLanguage();
+    }
+
+
+    /**
+     * this tests that the vanityURL proxies requests that are made to different hosts.
+     * In this case, we will request a url from dotcms and check to see that we get the results from dotcms.com
+     */
+
+    @Test
+    public void test_that_vanity_url_filter_handles_proxy_requests() throws Exception {
```

Comment:
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)